src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -958,9 +956,18 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                                 goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
                             }
```

Comment:
I'm wondering if I should try to optimze line 954.
I couldn't find an AdvSimd equivalent method for `Sse41.TestZ`. The original method is defined as follows:

```
int _mm_testz_si128 (__m128i a, __m128i b)

Compute the bitwise AND of 128 bits (representing integer data) in a and b,
and set ZF to 1 if the result is zero, otherwise set ZF to 0.
Compute the bitwise NOT of a and then AND with b, and set CF to 1 if the result is zero,
otherwise set CF to 0. Return the ZF value.
```

Reply:
`AdvSimd.CompareTest` should be the closest match

> Compare bitwise Test bits nonzero (vector). This instruction reads each vector element in the first source SIMD&FP
register, performs an AND with the corresponding vector element in the second source SIMD&FP register, and if
the result is not zero, sets every bit of the corresponding vector element in the destination SIMD&FP register to one,
otherwise sets every bit of the corresponding vector element in the destination SIMD&FP register to zero.

Reply:
Thanks! If I am understanding the API description correctly, this would be the equivalent operation:
```cs
if (AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask).ToScalar() == 0)
```

Reply:
No, `AdvSimd.CompareTest` operates on each element individually while `Sse41.TestZ` treats its operands as int128 integer.
You would need some kind of reduction to get the result to 0th element(e.g. `MaxAcross` )

Reply:
@echesakovMSFT still pending to address.

Reply:
@carlossanlop I replied in https://github.com/dotnet/runtime/pull/39041/files#r454731684

Basically, what CompareTest does is to check that result of bitwise AND of any element data with nonAsciiDataMask is non zero. In other words, the result of such operation would be `(utf16Data[i] & nonAsciiUtf16DataMask[i]) != 0 ? ~0 : 0`.

The same functionality as Sse2.TestZ provides could be done by doing MinAcross on the result of CompareTest. If all comparison were false (i.e. all elements are zero), then MinAcross would return 0, otherwise (i.e. if any of the resulting elements is all ones), -1.

Reply:
Thanks for the explanation. Code updated.